chpl/chpl-service/src/main/java/gov/healthit/chpl/activity/UserMaintenanceActivityMetadataBuilder.java

```diff
@@ -0,0 +1,63 @@
+package gov.healthit.chpl.activity;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import gov.healthit.chpl.domain.activity.ActivityCategory;
+import gov.healthit.chpl.domain.activity.ActivityMetadata;
+import gov.healthit.chpl.domain.activity.UserMaintenanceActivityMetadata;
+import gov.healthit.chpl.dto.ActivityDTO;
+import gov.healthit.chpl.dto.auth.UserDTO;
+
+@Component("userMaintenanceActivityMetadataBuilder")
+public class UserMaintenanceActivityMetadataBuilder extends ActivityMetadataBuilder {
+    private static final Logger LOGGER = LogManager.getLogger(VersionActivityMetadataBuilder.class);
+    private ObjectMapper jsonMapper;
+
+    public UserMaintenanceActivityMetadataBuilder() {
+        super();
+        jsonMapper = new ObjectMapper();
+    }
+
+    @Override
+    protected void addConceptSpecificMetadata(final ActivityDTO dto, final ActivityMetadata metadata) {
+        if (!(metadata instanceof UserMaintenanceActivityMetadata)) {
+            return;
+        }
+        UserMaintenanceActivityMetadata userMetadata = (UserMaintenanceActivityMetadata) metadata;
+
+        UserDTO origUser = getUserDtoFromJson(dto.getOriginalData(), dto.getId());
+        UserDTO newUser = getUserDtoFromJson(dto.getNewData(), dto.getId());
+
+        if (newUser != null) {
+            parseMetaData(newUser, userMetadata);
+        } else if (origUser != null) {
+            parseMetaData(origUser, userMetadata);
+        }
+
+        userMetadata.getCategories().add(ActivityCategory.USER_MAINTENANCE);
+    }
+
+    private void parseMetaData(UserDTO user, UserMaintenanceActivityMetadata userMetadata) {
+        userMetadata.setEmail(user.getEmail());
+        userMetadata.setSubjectName(user.getSubjectName());
+    }
+
+    private UserDTO getUserDtoFromJson(String json, Long dtoId) {
+        UserDTO userDTO = null;
+        try {
+            if (json != null && json != "") {
+                jsonMapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
```

Comment:
I had handled this issue in a different way with a @JsonIgnoreProperties annotation on the DTO objects themselves, but just wanting to check that you tested this with user activity from like back at the beginning of CHPL and know that it works. It would probably be best to handle this issue in the same way going forward for all activity types... this way is fine with me if you think it's better. If that's what we want to do can you remove that annotation that I had used and modify the other builders to do it this way? Alternatively switch to the way I had done it originally. Fine with either, just want it to be consistent.